packages/core/src/internal/utils/validators.ts

```diff
@@ -119,8 +119,16 @@ export function validateTopK(topK: number) {
 }
 
 export function validateTtlMinutes(ttlMinutes: number) {
-  if (ttlMinutes < 0) {
-    throw new InvalidArgumentError('ttlMinutes must be positive');
+  if (ttlMinutes < 0 || !Number.isInteger(ttlMinutes)) {
+    throw new InvalidArgumentError('ttlMinutes must be a positive integer');
+  }
+}
+
+export function validateTtl(ttl: number) {
```

Comment:
Since we have a methodName called `validateTtlMinutes`, I'd call this `validateTtlSeconds` just for parity with units being present in the name.